TileCanvas.java

```diff
@@ -0,0 +1,137 @@
+/**
+ * GamePanel is responsible for rendering the game board and the current score.
+ * It handles drawing the tiles with their corresponding colors and values.
+ * Owner -> TANUJ YADAV
+ * Date -> 17/09/2024
+ */
+
+import javax.swing.JPanel;
+import javax.swing.Timer;
+import java.awt.Graphics;
+import java.awt.Graphics2D;
+import java.awt.Color;
+import java.awt.Font;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+public class TileCanvas extends JPanel {
+    private int[][] gameBoard;
+    private int gridSize;
+    private int currentScore;
+    private boolean isAnimating;
+    private int animationStep;
+    private Timer animationTimer;
+    public TileCanvas(int[][] gameBoard, int gridSize, int currentScore) {
+        this.gameBoard = gameBoard;
+        this.gridSize = gridSize;
+        this.currentScore = currentScore;
+        this.isAnimating = false;
+        this.animationStep = 0;
+        animationTimer = new Timer(10, new ActionListener() {
+            public void actionPerformed(ActionEvent e) {
+                animationStep++;
+                if (animationStep > 20) {
+                    isAnimating = false;
+                    animationTimer.stop();
+                }
+                repaint();
+            }
+        });
+    }
+    protected void paintComponent(Graphics g) {
+        super.paintComponent(g);
+        Graphics2D graphics2D = (Graphics2D) g;
+        drawGameBoard(graphics2D);
+        drawScore(graphics2D);
+    }
+    /**
+     * Draws the game board and the tiles on it.
+     */
+    private void drawGameBoard(Graphics2D graphics2D) {
+        graphics2D.setColor(Color.BLACK);
+        graphics2D.fillRect(0, 0, getWidth(), getHeight() - 50);
+        int tileSize = (getWidth() - 20) / gridSize;
+        graphics2D.setFont(new Font("Arial", Font.BOLD, tileSize / 3));
+        for (int row = 0; row < gridSize; row++) {
+            for (int col = 0; col < gridSize; col++) {
+                drawTile(graphics2D, row, col, tileSize);
+            }
+        }
+    }

```

Comment:
Do not use documentation for defining the methods. Instead use Single Line Comments